commands/client.go

```diff
@@ -198,6 +202,48 @@ format is specified with the --enc flag.
 	},
 }
 
+// VerifyStorageDealResult wraps the success in an interface type
+type VerifyStorageDealResult struct {
+	validPip bool
+}
+
+var clientVerifyStorageDealCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Verify a storage deal",
+		ShortDescription: `
+Returns an error if the deal is not in the Complete state or the Piece Inclusion Proof
+is invalid.  Returns nil otherwise.
+`,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("id", true, false, "CID of deal to query"),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		proposalCid, err := cid.Decode(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+
+		resp, err := GetStorageAPI(env).QueryStorageDeal(req.Context, proposalCid)
+		if err != nil {
+			return err
+		}
+
+		if resp.State != storagedeal.Complete {
+			return errors.New("storage deal not in Complete state")
+		}
+
+		validateError := GetPorcelainAPI(env).ClientValidateDeal(req.Context, proposalCid, resp.ProofInfo)
+
+		if err := re.Emit(VerifyStorageDealResult{validateError == nil}); err != nil {
+			return err
+		}
+
+		return err
```

Comment:
The context of this err value is not apparent, just returning `re.Emit(...)` will help make this easier to read.

Reply:
Thanks, that was unintentional; the result of code thrashing.